### PR TITLE
Unifying duplicate msbuild properties in CoreCLR build

### PR DESCRIPTION
**Pull request by jkotas**
- __TargetOS -> TargetOS
- __BuildType, BuildType -> Configuration
- __BuildArch, BuildArch -> TargetArchitecture

### CODE REVIEW

File: src/coreclr/dir.common.props [hunk 1]
```diff
@@ -10,25 +10,18 @@
...
-    <BuildType Condition="'$(BuildType)'==''">Debug</BuildType>
-    <BuildType Condition="'$(BuildType)' == 'debug'">Debug</BuildType>
-    <BuildType Condition="'$(BuildType)' == 'release'">Release</BuildType>
-    <BuildType Condition="'$(BuildType)' == 'checked'">Checked</BuildType>
+    <Configuration Condition="'$(Configuration)'==''">Debug</Configuration>
+    <Configuration Condition="'$(Configuration)' == 'debug'">Debug</Configuration>
+    <Configuration Condition="'$(Configuration)' == 'release'">Release</Configuration>
+    <Configuration Condition="'$(Configuration)' == 'checked'">Checked</Configuration>
 
-    <!-- TODO: converge on one property for TargetOS and __TargetOS (and similar), and remove these extra lines. -->
```
Comment (jkotas) [inline src/coreclr/dir.common.props]: This is the TODO that made me to do this.

File: src/coreclr/build-runtime.cmd [hunk 2]
```diff
@@ -223,7 +223,7 @@ if %__BuildTypeDebug%==1    set __BuildType=Debug
 if %__BuildTypeChecked%==1  set __BuildType=Checked
```
Comment (jkotas) [inline src/coreclr/build-runtime.cmd]: I am not renaming any variables in shell scripts to make the change manageable.

**Comment by jkotas** [comment]
cc @dotnet/runtime-infrastructure

**Review by trylek — APPROVED** [review]
Awesome cleanup, thanks Jan for doing this!

**Comment by jkotas** [comment]
The test failure is #28553

**Review by ViktorHofer — APPROVED** [review]
Thanks

**Review by safern — COMMENTED** [review]
LGTM. Thanks.

**Comment by RobertHenry6bev** [comment]
This is a breaking change. Running on x64 ubuntu linux 19.1

Following https://github.com/dotnet/runtime/blob/master/docs/workflow/building/coreclr/linux-instructions.md#build-the-runtime-and-systemprivatecoreli and being a little more explicit about defaults, and using the latest clang drop:

Running 
bash ./build.sh \
          -subsetCategory coreclr \
          --configuration Debug \
          --clang10

Results in a malformed subdirectory name:
ls artifacts/bin/coreclr
Linux..Debug  Linux.x64.Debug
The Linux..Debug name is incorrect.  I suspect it should be Linux.x64.Debug, as it is in the commit immediately prior to this one.

**Comment by safern** [comment]
Thanks, @RobertHenry6bev, we just fixed here: https://github.com/dotnet/runtime/commit/b85563dd4819d69ba74007cf5e2bfc8ed4882163

And also, we have: https://github.com/dotnet/runtime/pull/34295 which defaults it in MSBuild.

If you fetch latest you shouldn't be hitting this issue anymore. 😄

**Pull request merged**